R/matching.R

```diff
@@ -88,6 +95,30 @@ estimate_weights <- function(data, centered_colnames = NULL, start_val = 0, meth
   wt <- exp(EM %*% alpha)
   wt_rs <- (wt / sum(wt)) * nrow(EM)
 
+  # bootstrapping
+  if (!is.null(nr.boot.iteration)) {
+    set.seed(set.seed.boot)
```

Comment:
@hoppanda I think we need to bit a careful with setting the seed here if we want to get on CRAN. We can try to use something like this pattern: 
```r
boot.array <- function(boot.out) {
  genv <- globalenv()

  # Make sure to leave '.Random.seed' as-is on exit
  old_seed <- genv$.Random.seed
  on.exit(suspendInterrupts({
    if (is.null(old_seed)) {
      rm(".Random.seed", envir = genv, inherits = FALSE)
    } else {
      assign(".Random.seed", value = old_seed, envir = genv, inherits = FALSE)
    }
  }))

  # Assign saved seed from boot.out
  assign(".Random.seed", value = boot.out$seed, envir = genv)

  # Generate same random numbers from boot() call
}
```